core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveDeltaIntegralCodec.java

```diff
@@ -103,9 +137,33 @@ protected ColumnPageEncoderMeta getEncoderMeta(ColumnPage inputPage) {
       protected List<Encoding> getEncodingList() {
         List<Encoding> encodings = new ArrayList<>();
         encodings.add(Encoding.ADAPTIVE_DELTA_INTEGRAL);
+        if (isInvertedIndex) {
+          encodings.add(Encoding.INVERTED_INDEX);
+        }
         return encodings;
       }
 
+      @Override
+      protected void fillLegacyFields(DataChunk2 dataChunk) throws IOException {
```

Comment:
can u please move this code to some utility and as below classes also has same logic